src/world/area_kkj/kkj_19/AF7C00.c

```diff
@@ -4,29 +4,273 @@
 
 #include "world/common/AddPlayerHandsOffset.inc.c"
 
+extern s32 D_80241B10_AF9460;
+extern s32 D_80241B14_AF9464;
+
+// Needs data migration, matching otherwise
+#ifdef NON_MATCHING
+ApiStatus func_802404DC_AF7E2C(Evt* script, s32 isInitialCall) {
+    Bytecode *args = script->ptrReadPos;
+
+    if (isInitialCall) {
+        D_80241B10_AF9460 = 0;
+    }
+
+    if (D_80241B10_AF9460 != 0) {
+        D_80241B10_AF9460 = 0;
+        evt_set_variable(script, *args, D_80241B14_AF9464);
+        return ApiStatus_DONE2;
+    }
+
+    return ApiStatus_BLOCK;
+}
+#else
 INCLUDE_ASM(s32, "world/area_kkj/kkj_19/AF7C00", func_802404DC_AF7E2C);
+#endif
+
+ApiStatus func_80240530_AF7E80(Evt* script, s32 isInitialCall) {
+    D_80241B14_AF9464 = evt_get_variable(script, *script->ptrReadPos);
+    D_80241B10_AF9460 = 1;
+
+    return ApiStatus_DONE2;
+}
+
+extern s32 D_802461F0[];
+
+ApiStatus func_80240568_AF7EB8(Evt* script, s32 isInitialCall) {
+    s32* ptr = (s32*)evt_get_variable(script, *script->ptrReadPos);
```

Comment:
instead of `*script->ptrReadPos`, we're trying to migrate all script functions that use ptrReadPos to have an `args` temp like you did above, and use that instead:

`Bytecode* args = script->ptrReadPos;`
`s32* ptr = (s32*)evt_get_variable(script, *args++);`